app/tools/variable_evaluator.rb

```diff
@@ -2,31 +2,27 @@
 
 class VariableEvaluator
   class << self
-    def evaluate_obj(obj, mentor_title = 'begeleider', mentor_gender = nil,
-                     student_name = 'deze student', student_gender = nil)
-      if obj.is_a?(String)
-        return evaluate(obj, mentor_title, mentor_gender,
-                        student_name, student_gender)
-      elsif obj.is_a?(Hash)
+    def evaluate_obj(obj, subs_hash)
+      return evaluate(obj, subs_hash) if obj.is_a?(String)
+      if obj.is_a?(Hash)
         obj.each do |k, v|
-          obj[k] = evaluate_obj(v, mentor_title, mentor_gender,
-                                student_name, student_gender)
+          obj[k] = evaluate_obj(v, subs_hash)
         end
       elsif obj.is_a?(Array)
         obj.each_with_index do |v, i|
-          obj[i] = evaluate_obj(v, mentor_title, mentor_gender,
-                                student_name, student_gender)
+          obj[i] = evaluate_obj(v, subs_hash)
         end
       end
       obj
     end
 
     private
 
-    def evaluate(text, mentor_title = 'begeleider', mentor_gender = nil,
-                 student_name = 'deze student', student_gender = nil)
-      mentor_title = 'begeleider' if mentor_title.blank?
-      substitutions_hash = substitutions(mentor_title, mentor_gender, student_name, student_gender)
+    def evaluate(text, subs_hash)
+      full_subs_hash = default_subs_hash.dup.merge(subs_hash.dup) do |_, oldval, newval|
+        newval.blank? ? oldval : newval
```

Comment:
Hoewel ik dit misschien duidelijker vindt, is hier niet een standaard ruby / rails manier voor? Bijv deepmerge? En dan net andersom?:

```ruby
[1] pry(main)> default = {a: 1, b: 2, c: 3}
=> {:a=>1, :b=>2, :c=>3}
[2] pry(main)> other = {c: 4, d: 5}
=> {:c=>4, :d=>5}
[3] pry(main)> default.merge(other)
=> {:a=>1, :b=>2, :c=>4, :d=>5}
[4] pry(main)> default.deep_merge(other)
=> {:a=>1, :b=>2, :c=>4, :d=>5}
[5] pry(main)> other.merge(default)
=> {:c=>3, :d=>5, :a=>1, :b=>2}
```

Reply:
nee dat gaat niet goed omdat als we wel een default waarde hebben willen we niet dattie bv door nil kan worden overschreven. Ook zitten we hier niet met geneste structuren.